File: Python/TREE/Identical Trees.py
//https://www.geeksforgeeks.org/problems/determine-if-two-trees-are-identical/1
class Solution:
    #Function to check if two trees are identical.
    def isIdentical(self,r1, r2):
        # Code here
        if r1 is None and r2 is None:
            return True
        if r1 is None or r2 is None:
            return False
        if r1.data != r2.data:
            return False
        return self.isIdentical(r1.left, r2.left) and self.isIdentical(r1.right, r2.right)

File: Java/TREE/Identical Trees.java
//https://www.geeksforgeeks.org/problems/determine-if-two-trees-are-identical/1
class Solution {
    // Function to check if two trees are identical.
    boolean isIdentical(Node r1, Node r2) {
        // Code Here
        if(r1==null && r2==null){
           return true;
       }
       if(r1==null && r2!=null){
           return false;
       }
       if(r1!=null && r2==null){
           return false;
       }
       if(r1.data!=r2.data){
           return false;
       }
       return isIdentical(r1.left,r2.left) && isIdentical(r1.right,r2.right);
    }
}
